### DISCUSSION TITLE
Config file & default language

### DISCUSSION THREAD

**Discussion by noraj** [tldr-pages/tldr#10796] [Q&A] [answered]
Is there a configuration file we can put for example in `~/.config/tldr/` to specify some default behavior?

For example, I would like to not have to provide `-L fr` at each command and to define a fallback to `en` when the page does not exist in French.

So I would expect to have a configuration file as below.

`~/.config/tldr/config.yml`

```yml
languages: fr,en
```

The config file would be nice to override the color or source as well.

Note: [tealdeer](https://github.com/dbrgn/tealdeer) has configuration support, see https://dbrgn.github.io/tealdeer/config.html, but not for languages tho (https://github.com/dbrgn/tealdeer/issues/251)

**Comment by kbdharun** [comment, marked as answer]
We do support having a configuration file in our official [Node client](https://github.com/tldr-pages/tldr-node-client?tab=readme-ov-file#configuration), [Rust client](https://github.com/tldr-pages/tlrc) and environmental variables in our [Python client](https://github.com/tldr-pages/tldr-python-client?tab=readme-ov-file#configuration) for customizing the language, platform, themes, repository, etc. [English is used as a default fallback when a page isn't present, so you can set only your]

The only exceptions for this are the [C client](https://github.com/tldr-pages/tldr-c-client) and [Extldr](https://github.com/tldr-pages/extldr) which are currently under-maintained (we are looking for new active maintainers for them, feel free to drop by the chatroom if you interested in maintaining it and let's have a discussion)

**Reply by noraj** [reply]
Python client no config file but reads some custom environment variables, node client reads a config file `.tldrrc` in a non-standard location, and the C and Elixir client have no global configuration support. So does that mean this is no tldr specification about a configuration file? It would be nice that all official clients read from the same standard file, eg `$XDG_CONFIG_HOME/tldr/config.toml`.

**Reply by kbdharun** [reply]
> Python client no config file but reads some custom environment variables, node client reads a config file `.tldrrc` in a non-standard location, and the C and Elixir client have no global configuration support.

Yes, the Python client reads it from the environment variables (forgot to update my comment, thanks).

> So does that mean this is no tldr specification about a configuration file? It would be nice that all official clients read from the same standard file, eg `$XDG_CONFIG_HOME/tldr/config.toml`.

Yes, currently there isn't a standard file that tldr clients read. There were some open (inactive) discussions about it at https://github.com/tldr-pages/tldr/issues/876 (I agree too that this is something that we should define in the specification), feel free to restart the discussion over there and also in our chatroom let's see what the other maintainers think.

But I am not sure how this would pan out in other platforms like Windows for instance since our official clients are cross-platform.

**Reply by ivanhercaz** [reply]
> Yes, currently there isn't a standard file that tldr clients read. There were some open (inactive) discussions about it at #876 (I agree too that this is something that we should define in the specification), feel free to restart the discussion over there and also in our chatroom let's see what the other maintainers think.

Having a standard for the configuration files specified in the clients specification would be awesome. I already thought about it for ExTldr but I really would prefer to follow a standard or the best practices in the most updated/used client.
